src/main/java/laddergame/controller/LadderController.java

```diff
@@ -31,19 +34,70 @@ public LadderController(final InputView inputView,
     }
 
     public void run() {
-        final Participants participants = retryOnRuntimeExceptionWithMessage(() -> new Participants(readNames()));
-        final Width ladderWidth = new Width(participants.getSize());
-        final Height ladderHeight = retryOnRuntimeExceptionWithMessage(() -> new Height(inputView.readHeight()));
-        final Ladder ladder = new Ladder(ladderWidth, ladderHeight, booleanGenerator);
+        final PersonalNames names = createPersonalNames();
+        final LadderResult ladderResult = createLadderResult(names);
+        final Width ladderWidth = new Width(names.getSize());
+        final Height ladderHeight = createHeight();
+
+        final Ladder ladder = makeLadder(ladderWidth, ladderHeight);
+
+        printLadder(names, ladderResult, ladder);
+        playGame(names, ladderResult, ladder);
+    }
 
+    private Ladder makeLadder(Width ladderWidth, Height ladderHeight) {
+        final LadderLinesGenerator ladderLinesGenerator = new LadderLinesGenerator(booleanGenerator);
+        final List<Line> lines = ladderLinesGenerator.createLines(ladderWidth, ladderHeight);
+        return new Ladder(ladderWidth, ladderHeight, lines);
+    }
+
+    private void playGame(PersonalNames names, LadderResult ladderResult, Ladder ladder) {
+        final LadderGame ladderGame = new LadderGame(names, ladderResult);
+        final GameResult result = ladderGame.moveAndGetResult(ladder);
+        printSearchedResultFrom(result, RETRY_LIMIT);
+    }
+
+    private void printLadder(PersonalNames names, LadderResult ladderResult, Ladder ladder) {
         final List<Line> lines = ladder.getLines();
-        outputView.printResult(ladderFormGenerator.generate(participants, lines));
+        outputView.printLadderForm(ladderFormGenerator.generate(names, ladderResult, lines));
+    }
+
+    private PersonalNames createPersonalNames() {
+        return createInstance(() -> new PersonalNames(inputView.readNames()), RETRY_LIMIT);
+    }
+
+    private LadderResult createLadderResult(PersonalNames names) {
+        return createInstance(() -> LadderResult.of(names, inputView.readResults()), RETRY_LIMIT);
+    }
+
+    private Height createHeight() {
+        return createInstance(() -> new Height(inputView.readHeight()), RETRY_LIMIT);
+    }
+
+    private <T> T createInstance(Supplier<T> supplier, int tryCount) {
+        while (tryCount-- > 0) {
+            try {
+                return supplier.get();
+            } catch (RuntimeException e) {
+                OutputView.printExceptionMessage(e.getMessage());
+            }
+        }
+        throw new IllegalStateException("재시도 횟수를 넘었습니다.");
+    }
+
+    private void printSearchedResultFrom(GameResult gameResult, int tryCount) {
+        while (tryCount-- > 0) {
+            String name = inputView.readNameToCheckResult();
+            if (isAll(name)) {
```

Comment:
isAll이라는 부분은 아직도 헷갈리네요 :)

Reply:
```suggestion
            if (isEnd(command)) {
```
이렇게 바꾸면 나을까요?

```java
private boolean isEnd(String name) {
    return name.equals("all");
}
```
반복의 종료임을 판단하는 책임을
저 메서드는 “all” 인지 확인하는 방식으로 수행한다고
생각해보았슴다,,, 🫥

Reply:
그렇다면, isEnt -> isFinish 등등으로 해도 좋곘네요 :) 
isEnd도 좋구요!